ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -162,18 +354,34 @@ public boolean equals(Object o) {
     if (!version.equals(blobId.version)) {
```

Comment:
`compareTo()` and `equals()` have to be consistent with each other, which it is here. But you can enforce it and simplify by simply doing `return compareTo(blobId) == 0;` at line 353 and avoiding the code duplication.